cachepool/test/test_cachepool.py

```diff
@@ -0,0 +1,327 @@
+            self.assertTrue(False)
+
+    def test_close_callback(self):
+
+        pool = cachepool.CachePool(
+                generator,
+                close_callback=close_callback,
+                )
+        element = pool.get()
+        pool.close(element)
+
+        self.assertTrue(element.closed)
+
+    def test_close_when_queue_full(self):
+
+        pool = cachepool.CachePool(
+                generator,
+                pool_size=1,
+                close_callback=close_callback,
+                )
+
+        dd('pool qsize: {0}'.format(pool.queue.qsize()))
+        element = pool.get()
+        pool.put(element)
+
+        dd('pool qsize: {0}'.format(pool.queue.qsize()))
+        element = generator()
+        pool.put(element)
+
+        dd('pool qsize: {0}'.format(pool.queue.qsize()))
+        self.assertTrue(element.closed)
+
+    def test_pool_size(self):
+
+        pool = cachepool.CachePool(
+                generator,
+                pool_size=1,
+                close_callback=close_callback,
+                )
+
+        element = pool.get()
+        pool.put(element)
+
+        element2 = Element()
+        pool.put(element2)
+
+        self.assertFalse(element.closed)
+        self.assertTrue(element2.closed)
+
+    def test_get_element(self):
+
+        pool = cachepool.CachePool(
+                generator,
+                )
+
+        element = pool.get()
+
+        dd('pool stat: {0}'.format(pool.stat))
+        self.assertEqual(1, pool.stat['new'])
+        self.assertEqual(1, pool.stat['get'])
+
+        pool.put(element)
+
+        dd('pool stat: {0}'.format(pool.stat))
+        self.assertEqual(1, pool.stat['put'])
+
+    def test_get_element_twice(self):
+
+        pool = cachepool.CachePool(
+                generator,
+                )
+
+        element = pool.get()
+
+        dd('pool stat: {0}'.format(pool.stat))
+        self.assertEqual(1, pool.stat['new'])
+
+        pool.put(element)
+        element = pool.get()
+
+        dd('pool stat: {0}'.format(pool.stat))
+        self.assertEqual(2, pool.stat['get'])
+
+    def test_args_argkw(self):
+
+        args = [1, 'x']
+        argkw = {'key': 'value'}
+
+        pool = cachepool.CachePool(
+                generator,
+                generator_args=args,
+                generator_argkw=argkw,
+                )
+
+        element = pool.get()
+
+        self.assertEqual(element.args[0], args[0])
+        self.assertEqual(element.args[1], args[1])
+        self.assertEqual(element.argkw, argkw)
+
+class TestCacheWrapper(unittest.TestCase):
+
+    def test_new(self):
+
+        pool = cachepool.CachePool(
+                generator,
+                )
+        wrapper = cachepool.CacheWrapper(
+                pool,
+                )
+
+        ele0 = ele1 = None
+        with wrapper as ele:
+            ele0 = ele
+
+            with wrapper as ele:
+                ele1 = ele
+
+        self.assertNotEqual(id(ele0), id(ele1))
```

Comment:
As I remember, there is a method `unittest.TestCase.assertIsNot` can do this better.
![image](https://cloud.githubusercontent.com/assets/44069/21542904/85e46618-cdfc-11e6-94a4-e8daa8da4b60.png)